tabular/src/autogluon/tabular/models/tab_transformer/tab_transformer_model.py

```diff
@@ -0,0 +1,469 @@
+                data = data.cuda()
+            with torch.no_grad():
+                data = Variable(data)
+                prob, _ = self.model(data)
+                batch_size = len(prob)
+                if self.problem_type != REGRESSION:
+                    prob = softmax(prob)
+
+            outputs[iter:(iter + batch_size)] = prob
+            iter += batch_size
+
+        if self.problem_type == BINARY:
+            return outputs[:, 1].cpu().numpy()
+        elif self.problem_type == REGRESSION:
+            outputs = outputs.flatten()
+            #return outputs.flatten().numpy()
+
+        return outputs.cpu().numpy()
+
+    def _get_default_searchspace(self):
+        return get_default_searchspace()
+
+
+    # # TODO: Do I need X_unlabeled here?
+    # def hyperparameter_tune(self, X_train, y_train, X_val, y_val, scheduler_options, **kwargs):
+    #
+    #     time_start = time.time()
+    #
+    #     self._set_default_searchspace()
+    #
+    #     scheduler_func = scheduler_options[0]
+    #     scheduler_options = scheduler_options[1]
+    #
+    #     # TODO: What are "scheduler_options" here ?
+    #     #trainloader = X_train.build_loader()
+    #     #valloader =
+    #
+    #
+    #     return self._get_hpo_results(scheduler=scheduler, scheduler_options=scheduler_options, time_start=time_start)
+
+
+
+    def save(self, file_prefix="", directory=None, return_filename=False, verbose=True):
+        """
+        file_prefix (str): Appended to beginning of file-name (does not affect directory in file-path).
+        directory (str): if unspecified, use self.path as directory
+        return_filename (bool): return the file-name corresponding to this save
+        """
+        import torch
+        if directory is not None:
+            path = directory + file_prefix
+        else:
+            path = self.path + file_prefix
+
+        params_filepath = path + self.params_file_name
+
+        os.makedirs(os.path.dirname(path), exist_ok=True)
+
+        if self.model is not None:
+            torch.save(self.model, params_filepath)
+
+        self.model = None  # Avoiding pickling the weights.
+        modelobj_filepath = super().save(path=path, verbose=verbose)
+
+        if return_filename:
+            return modelobj_filepath
+
+    @classmethod
+    def load(cls, path, file_prefix="", reset_paths=False, verbose=True):
+        """
+        file_prefix (str): Appended to beginning of file-name.
+        If you want to load files with given prefix, can also pass arg: path = directory+file_prefix
+        """
+        import torch
+        path = path + file_prefix
+        obj: TabTransformerModel = load_pkl.load(path=path + cls.model_file_name, verbose=verbose)
+        if reset_paths:
+            obj.set_contexts(path)
+
+        obj.model = torch.load(path + cls.params_file_name)
+
+        return obj
+
+        """
+        List of features to add (Updated by Anthony Galczak 10-20-20):
```

Comment:
0) enable output layer of TT model to be multiple fully connected layers rather than just a single linear layer.

Reply:
Added this to TODO list.